tests/llm_translation/test_cohere.py

```diff
@@ -487,5 +487,333 @@ def test_cohere_embed_v4_with_optional_params():
         assert all(isinstance(x, float) for x in response.data[0]['embedding'])
         assert isinstance(response.usage, litellm.Usage)
         
+    except Exception as e:
+        pytest.fail(f"Error occurred: {e}")
+
+
+# ==================== COHERE V2 API TESTS ====================
+
+@pytest.mark.parametrize("sync_mode", [True, False])
+@pytest.mark.asyncio
+@pytest.mark.flaky(retries=3, delay=1)
+async def test_cohere_v2_chat_completion(sync_mode):
+    """Test basic Cohere v2 chat completion functionality."""
+    try:
+        litellm.set_verbose = True
+        messages = [
+            {"role": "system", "content": "You are a helpful assistant."},
+            {"role": "user", "content": "Hello, how are you?"}
+        ]
+        
+        if sync_mode:
+            response = completion(
+                model="cohere_chat/v2/command-r-plus",
+                messages=messages,
+                max_tokens=50
+            )
+        else:
+            response = await litellm.acompletion(
+                model="cohere_chat/v2/command-r-plus",
+                messages=messages,
+                max_tokens=50
+            )
+        
+        # Validate response structure
+        assert response.choices is not None
+        assert len(response.choices) > 0
+        assert response.choices[0].message.content is not None
+        assert response.usage is not None
+        assert response.usage.total_tokens > 0
+        print(f"Cohere v2 response: {response}")
+        
+    except litellm.ServiceUnavailableError:
+        pass  # Skip if service is unavailable
+    except Exception as e:
+        pytest.fail(f"Error occurred: {e}")
+
+
```

Comment:
can you add a test "documents" and "citation_options" is getting sent in the request body